test/validation/api/ipsec/ipsec.c

```diff
@@ -516,6 +516,30 @@ static int send_pkts(const ipsec_test_part part[], int num_part)
 	return num_part;
 }
 
+/* Receive async inbound packet */
+static odp_event_t recv_pkt_async_inbound(odp_ipsec_op_status_t status)
+{
+	odp_event_t event;
+
+	if (status.error.sa_lookup == 0) {
+		do {
+			event = odp_queue_deq(suite_context.queue);
+		} while (event == ODP_EVENT_INVALID);
+
+		return event;
+	}
+
+	/*
+	 * In case of SA lookup failure, the event is enqueued to the default
+	 * queue specified during odp_ipsec_config()
+	 */
+	do {
+		event = odp_queue_deq(suite_context.default_queue);
+	} while (event == ODP_EVENT_INVALID);
```

Comment:
It would be better to use a temporary variable for the queue to be used so that the dequeuing loop would not need to be repeated. I think it would also help making the later patches of this PR just a little bit cleaner.

Reply:
Ack. Added queue variable in v5.